x-pack/plugins/aiops/public/embeddables/pattern_analysis/pattern_analysis_initializer.tsx

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiFlyoutHeader,
+  EuiTitle,
+  EuiFlyoutBody,
+  EuiForm,
+  EuiFormRow,
+  EuiSpacer,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyoutFooter,
+  EuiCallOut,
+} from '@elastic/eui';
+import { euiThemeVars } from '@kbn/ui-theme';
+import { i18n } from '@kbn/i18n';
+import type { FC } from 'react';
+import React, { useEffect, useMemo, useState, useCallback } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { pick } from 'lodash';
+import { isPopulatedObject } from '@kbn/ml-is-populated-object';
+import useObservable from 'react-use/lib/useObservable';
+import type { DataViewField } from '@kbn/data-views-plugin/public';
+import { useAiopsAppContext } from '../../hooks/use_aiops_app_context';
+import { DataSourceContextProvider } from '../../hooks/use_data_source';
+import type { PatternAnalysisEmbeddableRuntimeState } from './types';
+import { PatternAnalysisSettings } from '../../components/log_categorization/log_categorization_for_embeddable/embeddable_menu';
+import { RandomSampler } from '../../components/log_categorization/sampling_menu';
+import {
+  DEFAULT_PROBABILITY,
+  RANDOM_SAMPLER_OPTION,
+} from '../../components/log_categorization/sampling_menu/random_sampler';
+import type { MinimumTimeRangeOption } from '../../components/log_categorization/log_categorization_for_embeddable/minimum_time_range';
+import { getMessageField } from '../../components/log_categorization/utils';
+import { FieldSelector } from '../../components/log_categorization/log_categorization_for_embeddable/field_selector';
+import { SamplingPanel } from '../../components/log_categorization/sampling_menu/sampling_panel';
+
+export interface PatternAnalysisInitializerProps {
+  initialInput?: Partial<PatternAnalysisEmbeddableRuntimeState>;
+  onCreate: (props: PatternAnalysisEmbeddableRuntimeState) => void;
+  onCancel: () => void;
+  onPreview: (update: PatternAnalysisEmbeddableRuntimeState) => Promise<void>;
+  isNewPanel: boolean;
+}
+
+export const PatternAnalysisEmbeddableInitializer: FC<PatternAnalysisInitializerProps> = ({
+  initialInput,
+  onCreate,
+  onCancel,
+  onPreview,
+  isNewPanel,
+}) => {
+  const {
+    unifiedSearch: {
+      ui: { IndexPatternSelect },
+    },
+  } = useAiopsAppContext();
+
+  const [formInput, setFormInput] = useState<PatternAnalysisEmbeddableRuntimeState>(
+    pick(
+      initialInput ?? {
+        minimumTimeRangeOption: '1 week',
```

Comment:
The options have to have i18n support

Reply:
This is a good spot. I don't know how this got missed before, but all of these options are [not translated](https://github.com/elastic/kibana/blob/be39b67b5884076c5fba701574af17572423dede/x-pack/plugins/aiops/public/components/log_categorization/log_categorization_for_embeddable/minimum_time_range.ts).
I'll fix this for this PR

Reply:
Updated in bcfe91f2aeeb2807cd613c50e70184f852e7e181